atc/worker/gardenruntime/worker.go

```diff
@@ -715,6 +715,12 @@ func (worker *Worker) findOrStreamVolume(
 				return Volume{}, err
 			}
 
+			// TODO: find artifact again
+			// From `get` to where it artificat is used, it may take time, and during
+			// the period, original worker may have been pruned, but the artifact might
+			// have been streamed to other workers, thus find the artifact again can make
+			// sure the artifact exist.
```

Comment:
With my latest commit, this TODO is no longer needed. Even if original worker is pruned, worker resource cache will be marked as invalidated, but it can still be used by running builds.

But if the source worker is pruned, then the build may still fail, which is a general problem of volume streaming, I am not going to solve the problem in this PR.